### ISSUE TITLE
Paper’s Result is not reproducible

### ISSUE THREAD

**Issue by mandal4** [yanxp/MetaR-CNN#31]
As many issues in here said the results on the paper is not reproducible.
The performance shows high variance along the randomly sampled k-shot image in 2nd training phase. I think the model is something unstable for the task. 
I think it would be better the author release the sampled k-shot images description that is used to produce results on the paper as like another FSOD paper’s author on ICCV(2019) did. (Few-shot object detection via Feature reweighting)

**Comment by yanxp** [comment]
As there are variances of the novel samples of phase 2, we evaluate the model five times to get the average results as our paper. Because there are no standard novel sample setting before.

**Comment by Hxx2048** [comment]
@mandal4  Hello, I found the two paper are similar (FSOD and this paper）.
 Have you reproduce the results of FSOD?  I don't try FSOD because its code is based on python2.7.

**Comment by mandal4** [comment]
> @mandal4 Hello, I found the two paper are similar (FSOD and this paper）.
> Have you reproduce the results of FSOD? I don't try FSOD because its code is based on python2.7.

Yes i reproduce the results on that paper with very marginal gap. The author said evaluation on matlab reproduce the results. As i mentioned in this issues, there are fixed fine-tuning sample in that paper so the results were stable.

**Comment by Hxx2048** [comment]
> > @mandal4 Hello, I found the two paper are similar (FSOD and this paper）.
> > Have you reproduce the results of FSOD? I don't try FSOD because its code is based on python2.7.
> 
> Yes i reproduce the results on that paper with very marginal gap. The author said evaluation on matlab reproduce the results. As i mentioned in this issues, there are fixed fine-tuning samples in that paper so the results were stable.

I always felt that the idea of two papers are same (reweight meta featuers) but with different detectors(yolov2 and faster-rcnn). I would read codes in details again, thank you.

**Comment by Hxx2048** [comment]
> > @mandal4 Hello, I found the two paper are similar (FSOD and this paper）.
> > Have you reproduce the results of FSOD? I don't try FSOD because its code is based on python2.7.
> 
> Yes i reproduce the results on that paper with very marginal gap. The author said evaluation on matlab reproduce the results. As i mentioned in this issues, there are fixed fine-tuning samples in that paper so the results were stable.

I always felt that the idea of two papers are same (reweight meta featuers) but with different detectors(yolov2 and faster-rcnn). I would read codes in details again, thank you.

**Comment by Hxx2048** [comment]
@mandal4 In addition, do you use this version of the code? （FSOD）https://github.com/bingykang/Fewshot_Detection

**Comment by mandal4** [comment]
> As there are variances of the novel samples of phase 2, we evaluate the model five times to get the average results as our paper. Because there are no standard novel sample setting before.

I think, we could still select the subspace to be sampled to average. It's shame that the process is not written in your paper although the standard process is not fixed.

**Comment by mandal4** [comment]
> As there are variances of the novel samples of phase 2, we evaluate the model five times to get the average results as our paper. Because there are no standard novel sample setting before.

I think, we could still select the subspace to be sampled to average. It's shame that the process is not written in your paper although the standard process is not fixed.

**Comment by mandal4** [comment]
> @mandal4 In addition, do you use this version of the code? （FSOD）https://github.com/bingykang/Fewshot_Detection

Yes, i used that version of the code

**Comment by XiongweiWu** [comment]
@mandal4 To my point, reporting average score from randomly generated training sets is more reasonable, so I am not show whether the randomness also matters a lot in meta-yolo paper .

**Comment by NHW2017** [comment]
> As there are variances of the novel samples of phase 2, we evaluate the model five times to get the average results as our paper. Because there are no standard novel sample setting before.

Sorry to interrupt, I would like to ask if you have tried adding Focal loss or a loss function with similar functions to the code? How's the effect
